### ISSUE TITLE
Change Authentication method misbehavior in v3

### ISSUE THREAD

**Issue by JillieBeanSim** [zowe/zowe-explorer-vscode#3142]
**Describe the bug**

Error returned when logging in `Cannot read properties of undefined (reading 'includes'). This is likely caused by the extension that contributes zowe.profileManagement.`

Seeing use basic auth as an option when basic auth is in place and not seeing token auth option 
It says User & Password but in description mentions login
I see this is same as in v2 now but still confusing 

![Screenshot 2024-09-23 at 4 19 07 PM](https://github.com/user-attachments/assets/902eea13-d2b7-4ea6-b3e3-baab3c4d7244)

**Comment by traeok** [comment]
When testing with a `zosmf` profile type and logging in via API ML, I selected "User & Password" to generate the token. After generating the token I do not encounter any errors.

However, if I change my authentication method back to direct login using user/password, then an error occurs after entering in my password. The message is different than the one posted above but is still a TypeError:

![image](https://github.com/user-attachments/assets/002ccf23-7d9e-4217-abc4-8f7e57c88b2f)

---

Regarding the prompt that is mentioned in the original report: this prompt is intentional and was added once we implemented authentication for API ML when using PEM certificates. The prompt is not for doing a direct login with user & password - it is prompting for what authentication method you want to use to generate the token.
